### PR TITLE
10.0 improve ir rule website

### PR DESCRIPTION
**Pull request by iledarn**

### CODE REVIEW

File: ir_rule_website/doc/changelog.rst [hunk 1]
```diff
@@ -1,3 +1,8 @@
+`1.3.0`
+-------
+
+- **Improvement:** add ``website`` object to a rule evaluation context - to be able to use rules as such ``[('id','=', website.company_id.id)]``
```
Comment (yelizariev) [inline ir_rule_website/doc/changelog.rst]: It's a feature

File: ir_rule_website/models/ir_rule.py [hunk 2]
```diff
@@ -29,7 +29,9 @@ def _force_domain(self):
     @api.model
     def _eval_context(self):
         context = super(IrRule, self)._eval_context()
-        context['website_id'] = self._context.get('website_id') or self.env.user.backend_website_id.id
+        website_id = self._context.get('website_id') or self.env.user.backend_website_id.id
+        context['website_id'] = website_id
+        context['website'] = self.env['website'].browse(website_id)
```
Comment (yelizariev) [inline ir_rule_website/models/ir_rule.py]: Do you need a ``sudo()``? Can you ad some tests

Comment (iledarn) [reply ir_rule_website/models/ir_rule.py]: https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/ir/ir_rule.py#L123
domain_get method has sudo()

**Comment by iledarn** [comment]
@yelizariev , done

File: ir_rule_website/tests/test_backend_website_rule.py [hunk 3]
```diff
@@ -32,8 +35,41 @@ def setUp(self):
...
+
     def test_backend_website_rule(self):
         users = self.env['res.users'].sudo(user=self.user3.id).search([])
         self.assertTrue(self.user2 not in users)
         users = self.env['res.users'].sudo(user=self.user2.id).search([])
         self.assertTrue(self.user2 in users)
+
+        # case when in a rule the `website` object references another object (`res.company` in that case)
+        companies = self.env['res.company'].sudo(user=self.user1.id).search([])
+        self.assertFalse(self.company2 in companies)
```
Comment (yelizariev) [inline ir_rule_website/tests/test_backend_website_rule.py]: You can use ``assertNotIn``

**Pull request merged**